addons/web_editor/static/tests/link_test.js

```diff
@@ -0,0 +1,458 @@
+            inputText(".odoo-editor-editable p", "www.odoo.com");
+            insertText(editor, " ");
+            editor.clean();
+            await assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="http://www.odoo.com">www.odoo.com</a> </p>`
+            );
+        });
+
+        QUnit.test("typing odoo.com + space should convert to link", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            inputText(".odoo-editor-editable p", "odoo.com");
+            insertText(editor, " ");
+            editor.clean();
+            assert.strictEqual(
+                editable.innerHTML,
+                `<p><a href="http://odoo.com">odoo.com</a> </p>`
+            );
+        });
+
+        QUnit.test(
+            "typing http://odoo.com + space should convert to link",
+            async function (assert) {
+                const { editor, editable } = onMount();
+                const node = editable.querySelector("p");
+                setSelection(node, 0);
+                inputText(".odoo-editor-editable p", "http://odoo.com");
+                insertText(editor, " ");
+                editor.clean();
+                assert.strictEqual(
+                    editable.innerHTML,
+                    `<p><a href="http://odoo.com">http://odoo.com</a> </p>`
+                );
+            }
+        );
+
+        QUnit.test(
+            "typing http://google.co.in + space should convert to link",
+            async function (assert) {
+                const { editor, editable } = onMount();
+                const node = editable.querySelector("p");
+                setSelection(node, 0);
+                inputText(".odoo-editor-editable p", "http://google.co.in");
+                insertText(editor, " ");
+                editor.clean();
+                assert.strictEqual(
+                    editable.innerHTML,
+                    `<p><a href="http://google.co.in">http://google.co.in</a> </p>`
+                );
+            }
+        );
+
+        QUnit.test("typing www.odoo + space should not convert to link", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            inputText(".odoo-editor-editable p", "www.odoo");
+            insertText(editor, " ");
+            editor.clean();
+            assert.strictEqual(editable.innerHTML, "<p>www.odoo </p>");
+        });
+
+
+        QUnit.module("Toolbar based");
+
+        QUnit.test("should convert all selected text to link", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            insertText(editor, "Hello");
+            triggerEvent(node, "keydown", { key: "a", ctrlKey: true });
```

Comment:
Ok, this is clever. But I prefer that you use setSelection again instead.

Why?
This should work only when we have a single paragraph (it's the case, I know). But apparently it also works with multiple ones, and it shouldn't (if you select manually your lines, the create link button disappears, but ctrl+a seems to bypass it...)

Reply:
Surely, i thought, since it could be the way we can select, hence i wrote it that way, No issues i will update it.